### ISSUE TITLE
[BUG] Unable to add "is one of" filters with different capitalisations of the same value

### ISSUE THREAD

**Issue by jgough** [opensearch-project/OpenSearch-Dashboards#3428]
**Describe the bug**
If I add a filter for a field `is one of`: and then enter the same value twice with different capitalisations then the second one fails.
See the image below as it is easier to see rather than to explain

**To Reproduce**
1. Go to Discover
2. Click on Add Filter
3. Select an appropriate field
4. Choose "is one of"
5. Type "foo", press enter key.  It appears as a value
6. Type "Foo", press enter key
=> Says "**foo** has already been added"

**Expected behavior**
Expected that the value **Foo** appears next to **foo**

**OpenSearch Version**
2.5.0 *with a few custom-branding modifications*

**Dashboards Version**
2.5.0

**Screenshots**
![image](https://user-images.githubusercontent.com/968442/219065300-3f184b75-4fe9-4320-84c6-df9b4e8a94d6.png)

**Host/Environment (please complete the following information):**
 - Docker
 - Firefox 102.7.0esr

**Comment by joshuarrrr** [comment]
@jgough Yeah, I can confirm this buggy behavior. Related to https://github.com/opensearch-project/OpenSearch-Dashboards/issues/576, https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2251, and our inconsistent handling of case sensitivity in general.

**Comment by seraphjiang** [comment]
Thank you for reporting this issue. This issue has been open for a long time without recent activity, and OpenSearch Dashboards has since moved to the 3.x release line (currently 3.6, with 3.7 upcoming). The last supported 2.x release is 2.19.x.

We're closing this as part of a housekeeping effort to keep the backlog actionable. If this issue is still reproducible on a supported version (2.19.x or 3.x), please feel free to reopen this issue or file a new one with updated reproduction steps and version details.

We appreciate your contribution to OpenSearch!
